head.go

```diff
@@ -480,7 +480,7 @@ func (h *Head) Init(minValidTime int64) error {
 		return nil
 	}
 	level.Warn(h.logger).Log("msg", "encountered WAL error, attempting repair", "err", err)
-
+	sr.Close() // Close the reader so that the repair can remove the corrupted file under Windows.
```

Comment:
_My 2 cents from someone who's not so familiar with the tsdb code_

From an external view, having `defer sr.Close()` at L476 and `sr.Close()` at L483 feels a bit strange though I get that calling `Close()` isn't a problem per se. Would this work instead?

```go
	sr, err := wal.NewSegmentsRangeReader(wal.SegmentRange{Dir: h.wal.Dir(), First: startFrom, Last: -1})
	if err != nil {
		return errors.Wrap(err, "open WAL segments")
	}

	err = h.loadWAL(wal.NewReader(sr))
        sr.Close() // Close the reader so that the repair can remove the corrupted file under Windows.
	if err == nil {
		return nil
        }
```

And shouldn't the segment reader opened at L457 be closed ASAP instead of `defer`?

Reply:
yep that would work as well, Thanks!